sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -811,40 +905,55 @@ def fit(self, X, y=None):
                 " Please select a different metric."
             )
 
-        if self.algorithm != "auto":
+        if algorithms != {"auto"}:
```

Comment:
I think I would rather treat the default `mst_algorithm == algorithm == auto` case first and then disjoint on cases.

What do you think?